### DISCUSSION TITLE
Testing Machines

### DISCUSSION THREAD

**Discussion by seloner** [statelyai/xstate#1907] [Q&A]
Hello guys,how do you like to test your machines?Using Model Base testing with xstate/test or  something else ?
I noticed that xstate's source code is not using model base testing ?Why not ?

**Comment by ezhikov** [comment]
Hi. I'm now [experimenting in testing machine itself](https://github.com/davidkpiano/xstate/discussions/1904) (states and context). It is in very early stage, but I like how it is going so far. When I'll get to something not scrambled together from sticks and clay, I'll made dedicated repo.

**Comment by davidkpiano** [comment]
See the docs: https://xstate.js.org/docs/guides/testing.html
